### PR TITLE
os/mac: fix for major versions other than 10

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Just something I visually spotted.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Good catch @Bo98. Pushed two more I noticed.

File: Library/Homebrew/shims/super/cc [hunk 1]
```diff
@@ -188,7 +188,7 @@ class Cmd
     when "-Xpreprocessor", "-Xclang"
       # used for -Xpreprocessor -fopenmp
       args << arg << enum.next
-    when /-mmacosx-version-min=10\.(\d+)/
+    when /-mmacosx-version-min=\d+\.(\d+)/
       arg = "-mmacosx-version-min=10.9" if high_sierra_or_later? && $1.to_i < 9
```
Comment (Bo98) [inline Library/Homebrew/shims/super/cc:192]: Hmm sounds like this `if` condition will trigger for 11.0, which isn't what is meant to happen.

Comment (MikeMcQuaid) [reply Library/Homebrew/shims/super/cc:192]: I think it probably should happen there: it's bumping e.g. `-mmacosx-version-min=10.7` to `-mmacosx-version-min=10.9` on High Sierra or later. That seems correct on 11.0.

Comment (Bo98) [reply Library/Homebrew/shims/super/cc:192]: I'm talking about how `-mmacosx-version-min=11.0` would also be changed to `-mmacosx-version-min=10.9`.

Comment (MikeMcQuaid) [reply Library/Homebrew/shims/super/cc:192]: Ah, yes, lol. Fixing.

**Comment by MikeMcQuaid** [comment]
Thanks @Bo98!

**Pull request merged**